File: src/main/java/me/dreamhopping/pml/api/events/impl/client/net/ClientJoinServerEvent.java
package me.dreamhopping.pml.api.events.impl.client.net;

import me.dreamhopping.pml.api.events.Event;

import java.net.SocketAddress;

public class ClientJoinServerEvent extends Event {
    public final boolean isLocal;
    public final SocketAddress address;

    public ClientJoinServerEvent(boolean isLocal, SocketAddress address) {
        this.isLocal = isLocal;
        this.address = address;
    }
}

File: src/main/java/me/dreamhopping/pml/api/events/impl/client/net/ClientJoinServerEvent.kt
package me.dreamhopping.pml.api.events.impl.client.net

import me.dreamhopping.pml.api.events.Event
import java.net.SocketAddress

/**
 * Fired when the client has joined a server
 * @param isLocal if the server is hosted locally (a singleplayer world)
 * @param address the [SocketAddress] of the server
 */
class ClientJoinServerEvent(val isLocal: Boolean, val address: SocketAddress) : Event()